docs/source/spans.rst

```diff
@@ -0,0 +1,145 @@
+Spans
+=====
+.. currentmodule:: distributed
+
+.. note::
+
+    This is an experimental feature and may rapidly change without a deprecation cycle.
+
+Dask offers various diagnostics and :doc:`fine-performance-metrics` about tasks,
+grouping them by their prefix (commonly, the name of the function being called). This
+can be suboptimal:
+
+- your client code may be sophisticated enough that correlating lines on the client and
+  tasks being run on the cluster may prove difficult;
+- the same function may be applied to different parts of your workflow, with a different
+  performance profile;
+- you may be in a multitenancy setup, where part of the load on the cluster was not
+  originated by your client code.
+
+In these situations, it may be useful to attach arbitrary tags to your workflow, or
```

Comment:
They're arbitrary, but IMO the main selling point is that they can have meaning.
```suggestion
In these situations, it may be useful to attach meaningful tags to your workflow, or
```